node/connection_base.js

```diff
@@ -86,7 +86,13 @@ function connBaseRequest(options) {
     var req = self.buildOutRequest(options);
     self.ops.addOutReq(req);
     req.peer.invalidateScore();
+    req.errorEvent.on(onReqError);
+
     return req;
+
+    function onReqError(err) {
+        self.ops.popOutReq(req.id, req);
```

Comment:
also in that separate pr, please add a regression test, or have some existing test cover this prior bug so that we don't un-fix it ;-)